releasenotes/notes/fix-spin-op-index-order-024e76d21b08bb9a.yaml

```
@@ -0,0 +1,6 @@
+---
+fixes:
+  - |
+    Removes sign change in :meth:`qiskit_nature.second_q.operators.spin_op.index_order` carried over from
+    :class:`~qiskit_nature.second_q.operators.FermionicOp`'s implementation. Adds corresponding unit
+    tests to verify behavior.
```

```suggestion
    Removes wrong sign change from :meth:`qiskit_nature.second_q.operators.SpinOp.index_order`.
```
- fixes the method link
- removes the reference to `FermionicOp` as I think it is rather misleading here
- I do not think we need to mention the unittests as the end-user likely does not care about this development detail